tests/User/test_models.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+from django.core.exceptions import ValidationError
+from django.db.utils import DataError
+
+pytestmark=pytest.mark.django_db
+
+
+class TestUserModel:
+
+    def test_str_return(self, user_factory):
+        user = user_factory(email="shah@gmail.com")
+        assert user.__str__() == "shah@gmail.com"
+    
+    def test_valid_phone(self,user_factory):
+        user = user_factory(phone="0000000")
+        user.full_clean()
+    
+    def test_invalid_phone1(self, user_factory):
+        with pytest.raises(DataError):
+            user = user_factory(phone="+994700000000")
+            user.full_clean()
+
+    def test_invalid_phone2(self, user_factory):
+        with pytest.raises(ValidationError):
+            user = user_factory(phone="shahan")
+            user.full_clean()
```

Comment:
You should NEVER call internal functions, just create and save user, and see if error is raised